From 59e9f47342399e8b83e5108d9df4514eb8849fee Mon Sep 17 00:00:00 2001
From: Tobias Schramm <t.schramm@t-sys.eu>
Date: Mon, 12 Apr 2021 01:18:45 +0200
Subject: [PATCH 17/17] dt-binding: display: add documentation for
 startek,kd35wvfpd107

Previously dt documentation for the kd35wvfpd107 was missing. Fix that
by adding some

Signed-off-by: Tobias Schramm <t.schramm@t-sys.eu>
---
 .../display/panel/startek,kd35wvfpd107.yaml   | 76 +++++++++++++++++++
 1 file changed, 76 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/startek,kd35wvfpd107.yaml

diff --git a/Documentation/devicetree/bindings/display/panel/startek,kd35wvfpd107.yaml b/Documentation/devicetree/bindings/display/panel/startek,kd35wvfpd107.yaml
new file mode 100644
index 000000000000..5e8fdfb1c23f
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/startek,kd35wvfpd107.yaml
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/startek,kd35wvfpd107.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: kd35wvfpd107 16-bit parallel RGB panel
+
+maintainers:
+  - Tobias Schramm <t.schramm@manjaro.org>
+
+description: |
+  The kd35wvfpd107 is a 340x800 LCD panel with a 16-bit RGB interface.
+  It is based on the ST7701S controller, using 3-wire 9 bit SPI for
+  control.
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    items:
+      - const: startek,kd35wvfpd107
+
+  reg:
+    description: DSI virtual channel used by that screen
+    maxItems: 1
+
+  VCC-supply:
+    description: analog regulator for MIPI circuit
+
+  IOVCC-supply:
+    description: I/O system regulator
+
+  reset-gpios: true
+
+  backlight: true
+
+required:
+  - compatible
+  - reg
+  - VCC-supply
+  - IOVCC-supply
+
+additionalProperties: false
+
+examples:
+  - |
+   #include <dt-bindings/gpio/gpio.h>
+
+    spi-lcd {
+        compatible = "spi-gpio";
+        #address-cells = <1>;
+        ranges;
+
+        cs-gpios = <&pio 1 6 GPIO_ACTIVE_HIGH>; /* PB6 */
+        mosi-gpios = <&pio 1 9 GPIO_ACTIVE_HIGH>; /* PB9 */
+        sck-gpios = <&pio 1 7 GPIO_ACTIVE_HIGH>; /* PB7 */
+        num-chipselects = <1>;
+
+        rgb-panel@0 {
+            reg = <0>;
+            compatible = "startek,kd35wvfpd107";
+            backlight = <&backlight>;
+            pinctrl-names = "default";
+            pinctrl-0 = <&panel_pins>;
+            VCCIO-supply = <&vcc_3v3>;
+            VCC-supply = <&vcc_3v3>;
+
+            port {
+                lcd_in: endpoint {
+                    remote-endpoint = <&tcon0_out_lcd>;
+                };
+            };
+        };
+    };
-- 
2.30.1

